### ISSUE TITLE
Document the protocol used by Silence

### ISSUE THREAD

**Issue by keverets** [SilenceIM/Silence#689]
### Bug description

Though Silence states that it's based on the signal protocol ("A fork of Signal with only SMS/MMS encryption") that's insufficient for other projects to interoperate with Silence.

To that end, documenting the protocol used would allow for other implementations, or for it to be used in slightly different contexts.

The one under consideration (which motivated the submission of this issue) is JMP.chat which provides a gateway between SMS and XMPP. It would be beneficial for a user of that service to be able to have encrypted communication with Silence-using-contacts from their XMPP clients, rather than forcing that communication to be purely plain-text.

**Comment by whiver** [comment]
I totally agree, that joins my comment on this issue https://github.com/SilenceIM/Silence/issues/686#issuecomment-404931414 suggesting that more doc on the Silence architecture would be welcome.

Concerning XMPP, being able to connect an app that would add the support for XMPP would be way better in my sense than trying to implement it directly in Silence as tracked in #390, since we already lack of maintenance for the existing features, without adding this huge one...

**Comment by paulcmal** [comment]
@ArchangeGabriel mentioned the issue of documentation in #670 again. Can we agree this is a top priority for us as a Silence-users community and we should set up some form of working group to reach this goal?

This would help not only more people contribute to Silence, but also write interoperable (i.e. compatible) clients and services.

I guess somebody who knows the codebase could already start document the crypto primitives and magic bits used. I'm willing to give some time in a sort of hackathon to help out with this although my cryptographic knowledge is limited.
